### PR TITLE
Issue/3073

### LINKED ISSUE #3073: Support for 'to merge-datasets' transformation using data groups
# Context
Merge-datasets transformation is slightly different from the others. UI needs to specify a source dataset to be merged in current/target one besides selecting the merge-column in both datasets.
Also UI/transformation let the user select any column that user needs from source dataset.

# Problem or idea
How/Where to place new merged columns?

# Solution or next step
We decided to create same data-groups in target dataset as user has selected in source dataset UI

### PR DESCRIPTION
**Pull request by iperdomo**
Closes #3073

### CODE REVIEW

File: backend/src/akvo/lumen/lib/transformation/merge_datasets.clj [hunk 1]
```diff
@@ -141,24 +205,57 @@
...
+                                                  :aggregation-direction (get source "aggregationDirection")
+                                                  :spec source}
+                                         :target {:merge-column (get target "mergeColumn")
+                                                  :table-name (:table-name target-merge-data-group)}})
+                  (map #(to-sql-types % (walk/stringify-keys columns))))]
+        (postgres/create-dataset-table conn table-name (map #(update % :id (fn [_]
+                                                                             (:columnName %)))
+                                                            (walk/keywordize-keys columns)))
+        (jdbc/insert-multi! conn table-name data)))
+    #_(map #(dissoc % :original-table-name) data-groups-to-be-created)
```
Comment (tangrammer) [inline backend/src/akvo/lumen/lib/transformation/merge_datasets.clj:245]: should we leave this one or just remove it?

File: backend/src/akvo/lumen/lib/transformation/merge_datasets.clj [hunk 2]
```diff
@@ -118,14 +148,48 @@
...
-  (let [dataset-id (get source "datasetId")]
-    (if-let [data (aggregation.commons/table-name-and-columns-from-data-grops conn dataset-id)]
-      (update data :columns walk/stringify-keys)
-      nil
-      ;; TODO: implemenmt exception when migrated fully to data-groups
-      #_(throw (ex-info (format "Dataset %s does not exist" dataset-id)
-                      {:source source})))))
+(defn get-data-groups-to-be-created [conn source target-dataset-columns]
+  (let [dataset-id (get source "datasetId")
+        ;; TODO what would happen if dataset source doesn't have data-groups (old dataset => dataset-version-1)
```
Comment (tangrammer) [inline backend/src/akvo/lumen/lib/transformation/merge_datasets.clj:153]: we could remove these reminders now

File: backend/src/akvo/lumen/lib/transformation/merge_datasets.clj [hunk 3]
```diff
@@ -118,14 +148,48 @@
...
+                              (map-indexed (fn [i column-name]
+                                             (let [dg (engine/datagroup-by-column data-groups column-name)
+                                                   column (first (filter #(= (get % "columnName") column-name) (:columns dg)))]
+                                               (-> column
+                                                   (assoc "sourceColumnName" (get column "columnName")
+                                                          "columnName" (engine/derivation-column-name
+                                                                     (+ (engine/next-column-index target-dataset-columns) i)))
+                                                reset-column-values))))
+                              (group-by #(get % "groupId")))
+        instance-id-column (->> (db.data-group/get-data-group-by-column-name ;; TODO: define column as a var? (def instance-id-col {})
```
Comment (tangrammer) [inline backend/src/akvo/lumen/lib/transformation/merge_datasets.clj:172]: same question with this TODO line, should we keep or tackle it?

**Review by tangrammer — COMMENTED** [review]
a few small comments added

**Comment by iperdomo** [comment]
Let's remove those `TODO` for now

**Review by tangrammer — APPROVED** [review]

**Pull request merged**